api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -75,6 +75,14 @@ public R or(R leftResult, R rightResult) {
       return null;
     }
 
+    public <T> R isNaN(BoundReference<T> ref) {
+      return null;
+    }
+
+    public <T> R notNaN(BoundReference<T> ref) {
+      return null;
```

Comment:
I think new methods should throw `UnsupportedOperationException` so that visitors must be updated to work with `isNaN`.

Reply:
I initially made these methods abstract to ensure I implemented them in all the child classes, but `UnsupportedOperationException` is definitely a better idea!